sunkit_image/asda.py

```diff
@@ -11,410 +11,379 @@
-
-        Parameters
-        ----------
-        r : `float`, optional
-            Radius which defaults to 0.
-
-        Return
-        ------
-        `float`
-            Radial speed at the radius of ``r``.
-        """
-        r = r + 1e-10
-        return self.get_vtheta(r) * self.ratio_vradial
-
-    def get_vxvy(self, x_range, y_range, x=None, y=None):
-        """
-        Calculates the velocity field in a meshgrid generated with ``x_range``
-        and ``y_range``.
-
-        Parameters
-        ----------
-        x_range : `list`
-            Range of the x coordinates of the meshgrid.
-        y_range : `list`
-            range of the y coordinates of the meshgrid.
-        x, y : `numpy.meshgrid`, optional
-            If both are given, ``x_range`` and ``y_range`` will be ignored.
-            Defaults to None``.
-
-        Return
-        ------
-        `tuple`
-            The generated velocity field ``(vx, vy)``.
-        """
-        if len(x_range) != 2:
-            self.x_range = [0 - self.rmax, self.rmax]
-        if len(y_range) != 2:
-            self.y_range = [0 - self.rmax, self.rmax]
-        if x is None or y is None:
-            # Check if one of the input parameters is None but the other one is not None
-            if x != y:
-                warnings.warn("One of the input parameters is missing, setting both to 'None'", stacklevel=3)
-                x, y = None, None
-            # Creating mesh grid
-            x, y = self.get_grid(x_range=x_range, y_range=y_range)
-        # Calculate radius
-        r = np.sqrt(np.square(x) + np.square(y)) + 1e-10
-        # Calculate velocity vector
-        vector = [
-            0 - self.get_vtheta(r) * y + self.get_vradial(r) * x,
-            self.get_vtheta(r) * x + self.get_vradial(r) * y,
-        ]
-        self.vx = vector[0] / r
-        self.vy = vector[1] / r
-        return self.vx, self.vy
+
+def get_vtheta(gamma, rcore, r=0):
+    """
+    Calculate rotation speed at radius of ``r``.
+
+    Parameters
+    ----------
+    gamma : `float`, optional
+        A replacement for ``vmax`` and only used if both ``gamma`` and ``rcore``are not `None`.
+        Defaults to `None`.
+    rcore : `float`, optional
+        A replacement for ``rmax`` and only used if both ``gamma`` and ``rcore``are not `None`.
+        Defaults to `None`.
+    r : `float`, optional
+        Radius which defaults to 0.
+
+    Return
+    ------
+    `float`
+        Rotating speed at radius of ``r``.
+    """
+    r = r + 1e-10
+    return gamma * (1.0 - np.exp(0 - np.square(r) / np.square(rcore))) / (2 * np.pi * r)
+
+
+def get_vradial(gamma, rcore, ratio_vradial, r=0):
+    """
+    Calculate radial (expanding or shrinking) speed at radius of ``r``.
+
+    Parameters
+    ----------
+    gamma : `float`, optional
+        A replacement for ``vmax`` and only used if both ``gamma`` and ``rcore``are not `None`.

```

Comment:
```suggestion
        A replacement for ``vmax`` and only used if both ``gamma`` and ``rcore`` are not `None`.
```

And in the other lines as well.

Reply:
Thanks I believe I fixed it just a couple of minutes ago.